onnxruntime/core/providers/dml/OperatorAuthorHelper/OperatorHelper.cpp

```diff
@@ -923,6 +923,24 @@ namespace OperatorHelper
         const uint32_t inputDimCount = gsl::narrow_cast<int32_t>(inputDimensions.size());
         const uint32_t axis = operatorAttributes.GetOptionalAttribute<int32_t>(AttrName::Axis, 0);
         m_axis = static_cast<int>(HandleNegativeAxis(axis, inputDimCount));
+
+        if (opsetVersion >= 18) // num_outputs attribute is only defined in opset18.
+        {
+            const uint32_t numOutputs = operatorAttributes.GetOptionalAttribute<int32_t>(AttrName::NumOutputs, 0);
+            if (numOutputs > 0)
+            {
+                ML_CHECK_VALID_ARGUMENT(m_split.size() == 0);
+                auto splitDimSize = inputDimensions.at(m_axis);
+                auto evenParts = static_cast<int>(ceil((double)splitDimSize / numOutputs));
+                m_split = std::vector<int>(numOutputs, evenParts);
+                m_split.back() = static_cast<int>(splitDimSize - (numOutputs-1) * evenParts);
```

Comment:
> auto evenParts = static_cast<int>(ceil((double)splitDimSize / numOutputs));

It appears we don't already have a helper in ORT DML EP utility functions for `IntegerRoundUp` in `OperatorUtility.h` or `DmlCommon.h`, but it's likely to be useful in other cases, and it's just the following (no need for double precision):

```
uint32_t RoundUp(uint32_t x, uint32_t y)
{
    assert(y != 0);
    return (x + y - 1) / y;
}
```

```
DimensionType inputLength = inputDimensions.at(m_axis);
DimensionType outputPartLength = RoundUp(inputLength, outputCount);
m_split.resize(outputCount, outputPartLength);
m_split.back() = static_cast<int>(inputLength % outputPartLength);
```